components/org.wso2.carbon.identity.api.server.script.library/org.wso2.carbon.identity.api.server.script.library.v1/src/main/java/org/wso2/carbon/identity/api/server/script/library/v1/impl/ScriptLibrariesApiServiceImpl.java

```diff
@@ -0,0 +1,90 @@
+import org.wso2.carbon.identity.api.server.common.ContextLoader;
+import org.wso2.carbon.identity.api.server.script.library.v1.ScriptLibrariesApiService;
+import org.wso2.carbon.identity.api.server.script.library.v1.core.ServerScriptLibrariesService;
+import org.wso2.carbon.identity.api.server.script.library.v1.model.ScriptLibraryListResponse;
+import org.wso2.carbon.identity.api.server.script.library.v1.model.ScriptLibraryResponse;
+
+import java.io.InputStream;
+import java.net.URI;
+
+import javax.ws.rs.core.Response;
+
+import static org.wso2.carbon.identity.api.server.common.Constants.V1_API_PATH_COMPONENT;
+import static org.wso2.carbon.identity.api.server.script.library.common.Constants.SCRIPT_LIBRARY_PATH_COMPONENT;
+
+/**
+ * Implementation of the script library Rest API.
+ */
+public class ScriptLibrariesApiServiceImpl implements ScriptLibrariesApiService {
+
+    @Autowired
+    private ServerScriptLibrariesService serverScriptLibrariesService;
+
+    @Override
+    public Response addScriptLibrary(String name, InputStream contentInputStream, Attachment contentDetail,
+                                     String description) {
+
+        serverScriptLibrariesService.addScriptLibrary(name, contentInputStream, description);
+        URI location =
+                ContextLoader.buildURIForHeader(V1_API_PATH_COMPONENT + SCRIPT_LIBRARY_PATH_COMPONENT + "/" + name);
+        return Response.created(location).build();
+    }
+
+    @Override
+    public Response deleteScriptLibrary(String scriptLibraryName) {
+
+        serverScriptLibrariesService.deleteScriptLibrary(scriptLibraryName);
+        return Response.noContent().build();
+    }
+
+    @Override
+    public Response getScriptLibraries(Integer limit, Integer offset) {
+
+        ScriptLibraryListResponse scriptLibraryListResponse = serverScriptLibrariesService.getScriptLibraries(limit,
+                offset);
+        return Response.ok().entity(scriptLibraryListResponse).build();
+    }
+
+    @Override
+    public Response getScriptLibraryByName(String scriptLibraryName) {
+
+        ScriptLibraryResponse scriptLibraryResponse = serverScriptLibrariesService.getScriptLibrary(scriptLibraryName);
+        return Response.ok().entity(scriptLibraryResponse).build();
+    }
+
+    @Override
+    public Response getScriptLibraryContentByName(String scriptLibraryName) {
+
+        return Response.ok().entity(serverScriptLibrariesService.getScriptLibraryContentByName(scriptLibraryName))
+                .build();
+    }
+
+    @Override
+    public Response updateScriptLibrary(String scriptLibraryName, InputStream contentInputStream,
+                                        Attachment contentDetail, String description) {
+
+        ScriptLibraryResponse scriptLibraryResponse =
+                serverScriptLibrariesService.updateScriptLibrary(scriptLibraryName, contentInputStream, description);
+        return Response.ok().entity(scriptLibraryResponse).build();
```

Comment:
let's avoid sending any response in update requests as discussed in the code-review

Reply:
[Fixed  commit](https://github.com/wso2/identity-api-server/pull/81/commits/2800691d849b18ddbdf4fb3acff4c9f75b8e2795)